app/CreateLocations.tsx

```diff
@@ -1,17 +1,73 @@
-import { ReactElement } from "react";
+import { ReactElement, useState } from "react";
 import { Wrapper } from "./ui/Wrapper";
 import { Button } from "react-bootstrap";
 import { greenActiveButton, greyInactiveButton } from "./ui/colourLibrary";
 import { LargeButton } from "./ui/LargeButton";
+import { StyledForm } from "./ui/StyledForm";
+import { PageHeader } from "./ui/PageHeader";
+import { FormLabel } from "./ui/FormLabel";
+import { FormGroup } from "./ui/FormGroup";
+import { FormControl } from "./ui/FormControl";
 
 export function CreateLocations(props: {
   onCreateLocationClick: () => void;
   onCancelClick: () => void;
 }): ReactElement {
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const [displayName, setdisplayName] = useState("");
+  const [fullName, setFullName] = useState("");
+  const [suburb, setSuburb] = useState("");
+
   const { onCreateLocationClick, onCancelClick } = props;
   return (
     <Wrapper>
-      Create Locations
+      <PageHeader> Create Location</PageHeader>
+      <div>
+        <StyledForm>
+          <FormGroup>
+            <FormLabel>Display name</FormLabel>
+            <FormControl
+              type="text"
+              placeholder="enter display name of store"
+              value={displayName}
+              onChange={(e) => {
+                const newDisplayName = e.target.value;
```

Comment:
we seem to be doing this `e.target.value` thing a fair bit and will only do more as time goes one, can we find a way to avoid repeating it?

Reply:
we could just shorten it by not declaring the const newUsername and just going setUsername(e.target.value)

Reply:
or make a function in a utils file that replaces it?

Reply:
you can't use utils since this function relies on `setDisplayName` which is only available in the component. basically I just want to be able to pass `setDisplayName` (or whatever state-setting function) directly, like `onChange={setDisplayName}`

might need to do something similar to the button where we make our own version

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/15/commits/0c4f317b9e0ddf140938134d289e24211dcccf1c